plugins/inputs/t128_graphql/t128_graphql_test.go

```diff
@@ -0,0 +1,442 @@
+package t128_graphql_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	plugin "github.com/influxdata/telegraf/plugins/inputs/t128_graphql"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+type Endpoint struct {
+	URL             string
+	Code            int
+	ExpectedRequest string
+	Response        string
+}
+
+const (
+	ValidExpectedRequestNoTag     = `{"query":"query MyQuery{allRouters(name:\"ComboEast\"){nodes{nodes(name:\"east-combo\"){nodes{arp{nodes{\ntest-field\n}}}}}}}"}`
+	ValidQueryNoTag               = "query MyQuery{allRouters(name:\"ComboEast\"){nodes{nodes(name:\"east-combo\"){nodes{arp{nodes{\ntest-field\n}}}}}}}"
+	ValidExpectedRequestSingleTag = `{"query":"query MyQuery{allRouters(name:\"ComboEast\"){nodes{nodes(name:\"east-combo\"){nodes{arp{nodes{\ntest-field\ntest-tag\n}}}}}}}"}`
+	ValidQuerySingleTag           = "query MyQuery{allRouters(name:\"ComboEast\"){nodes{nodes(name:\"east-combo\"){nodes{arp{nodes{\ntest-field\ntest-tag\n}}}}}}}"
+	ValidExpectedRequestDoubleTag = `{"query":"query MyQuery{allRouters(name:\"ComboEast\"){nodes{nodes(name:\"east-combo\"){nodes{arp{nodes{\ntest-field\ntest-tag-1\ntest-tag-2\n}}}}}}}"}`
+	ValidQueryDoubleTag           = "query MyQuery{allRouters(name:\"ComboEast\"){nodes{nodes(name:\"east-combo\"){nodes{arp{nodes{\ntest-field\ntest-tag-1\ntest-tag-2\n}}}}}}}"
+	InvalidRouterExpectedRequest  = `{"query":"query MyQuery{allRouters(name:\"not-a-router\"){nodes{nodes(name:\"east-combo\"){nodes{arp{nodes{\ntest-field\n}}}}}}}"}`
+	InvalidRouterQuery            = "query MyQuery{allRouters(name:\"not-a-router\"){nodes{nodes(name:\"east-combo\"){nodes{arp{nodes{\ntest-field\n}}}}}}}"
+	InvalidFieldExpectedRequest   = `{"query":"query MyQuery{allRouters(name:\"ComboEast\"){nodes{nodes(name:\"east-combo\"){nodes{arp{nodes{\ninvalid-field\n}}}}}}}"}`
+	InvalidFieldQuery             = "query MyQuery{allRouters(name:\"ComboEast\"){nodes{nodes(name:\"east-combo\"){nodes{arp{nodes{\ninvalid-field\n}}}}}}}"
+)
+
+var ResponseProcessingTestCases = []struct {
+	Name            string
+	EntryPoint      string
+	Fields          map[string]string
+	Tags            map[string]string
+	Query           string
+	Endpoint        Endpoint
+	ExpectedMetrics []*testutil.Metric
+	ExpectedErrors  []string
+}{
+	{
+		Name:       "empty query produces error",
+		EntryPoint: "",
+		Fields:     nil,
+		Tags:       nil,
+		Query:      "query MyQuery{{\n}}",
+		Endpoint: Endpoint{"/api/v1/graphql", 400, `{"query":"query MyQuery{{\n}}"}`, `
+		{
+			"errors": [{
+				"name": "BadRequestError",
+				"message": "Must provide query string."
+			}]
+		}
+		`},
+		ExpectedMetrics: nil,
+		ExpectedErrors: []string{
+			"status code 400 not OK for metric test-metric: Must provide query string.",
```

Comment:
We shouldn't even be making a query for an empty entryPoint in the first place. There is no need to bother nginx/node etc with this if you know what I mean.

Reply:
Agreed, I'll handle this in `Init`

Reply:
- [ ] update test after handling in Init